### DISCUSSION TITLE
Mods able to OE without permissions

### DISCUSSION THREAD

**Discussion by bertboy13** [Jikoo/OpenInv#266] [Q&A]
We have an issue where mods with the permissions openinv.enderchest.edit set to false are still able to remove items from peoples enderchest - any hints on a fix, perhaps we've done something incorrectly. 
![Screenshot 2024-12-23 195253](https://github.com/user-attachments/assets/89aa9902-0cc0-48c1-bb1f-47405b571e50)

**Comment by Jikoo** [comment]
The easiest way to test your permissions is to enable verbose mode with LuckPerms. If you filter for permissions starting with `openinv`, you can see what permissions OI is checking and how they evaluate. Note that perms for edit-only modes are only evaluated when the inventory is opened - the opened inventory will contain display-only slots in place of normal slots.

**Reply by Litning11** [reply]
Hey @Jikoo, I work on the same server. 

I think there may have been a miscommunication, we do know what permissions have been given. In this case `openinv.enderchest.open.other` is true, but `openinv.enderchest.edit` is false. Despite this, people with the ability to open enderchests are still able to edit even though they shouldn't be able to. This is a suspected bug.

**Reply by Jikoo** [reply]
The fact that the permissions are denied in one place doesn't mean that they aren't granted in another. Permissions plugins support tracks, individual permissions, multiple inheritance, and a whole lot of mess that can very often lead to configurations that appear to be correct at a glance not functioning as expected.

I get that you guys think your permissions are configured correctly. Please use the debug tools at your disposal to check - it will take you maybe 5 minutes tops versus me having to play guessing games for hours to try to replicate potential problems.

**Reply by Litning11** [reply]
![image](https://github.com/user-attachments/assets/b153587b-3ab2-469d-9700-de4e012f4a4e)

![image](https://github.com/user-attachments/assets/7a3268d9-849e-4eb3-81e6-d8fb65f06b1d)

Here are the results of the verbose check, I'm not sure how that provides any information since I don't know how to check them 'editing' something as that is done through the GUI, not command. The second image is the return on me filtering the entire server permission structure for instances of openinv, here you can see we only have the following permissions under the Mod group and nowhere else.

**Reply by Jikoo** [reply]
Could you please repeat the debug test by enabling it, running the command, and then disabling it? OI uses 2 tasks during the course of execution, and the view-only checks come at the end of the tasks. The way you are using it is causing it to report the initial checks to use the command, not the checks that occur during execution once the target is determined.

**Reply by Litning11** [reply]
Okay, I ran a verbose profiler and had them open up a random EC and move some items around. These are the results: https://luckperms.net/verbose/vUa0tilBR8

It's a bit long so I'll grab some snippets where OE show up: 
![image](https://github.com/user-attachments/assets/7023d5cb-0703-431a-b079-d463e2bed5f9)

![image](https://github.com/user-attachments/assets/e8101739-067b-4864-85a6-d844ee9fc796)

**Reply by Jikoo** [reply]
Okay, thanks, that's super helpful. Love that the upload has stack traces in it, I'm used to the ingame one. Right before the access level check we can see that there's a check for opening other, which should be the edit permission.
![image](https://github.com/user-attachments/assets/93dd7541-4502-46a3-957b-4c03429e1a52)
Sure enough, a silly mistake on my end:
https://github.com/Jikoo/OpenInv/blob/56f8cf4d7138646153977ed1f4a0b59766283d8f/plugin/src/main/java/com/lishid/openinv/OpenInv.java#L253

**Reply by Jikoo** [reply]
I've pushed f5c31c32d390d866a4a370a4aa7905d220225a67, which can be downloaded from https://github.com/Jikoo/OpenInv/actions/runs/12615837548 (the dist artifact) when it finishes. I will probably cut a release in the next few days as well.

**Reply by Litning11** [reply]
No problem! We'll probably wait for the full release as I just personally avoid running dev builds on production, even if they're fully safe lol. Happy to have helped.

**Reply by Litning11** [reply]
@Jikoo We did end up using the dev build and are now facing this issue: 

![image](https://github.com/user-attachments/assets/57549f75-20e1-452d-be1c-7754b66b421c)

Settings are completely unchanged since previously reported, except now Mods are completely unable to view EnderChests.

**Reply by Jikoo** [reply]
Please update to 5.1.6.
